test/unit_test/admin/conftest.py

```
@@ -0,0 +1,298 @@
+    stubs["api.db.services"] = api_db_services
+
+    user_service_mod = types.ModuleType("api.db.services.user_service")
+    user_service_mod.TenantService = MagicMock(name="TenantService")
+    user_service_mod.UserTenantService = MagicMock(name="UserTenantService")
+    stubs["api.db.services.user_service"] = user_service_mod
+
+    stubs["api.db"] = api_db
+
+    # Cross-link parent → child attributes so ``from api.X import Y``
+    # resolves during import.
+    setattr(api, "common", api_common)
```

Ruff (0.15.21)

[warning] Do not call `setattr` with a constant attribute value. It is not any safer than normal property access.

Replace `setattr` with assignment

(B010)